lib/puppet/pops/types/type_factory.rb

```diff
@@ -181,13 +181,7 @@ def self.any
   # @api public
   #
   def self.regexp(pattern = nil)
-    if pattern
-      t = PRegexpType.new(pattern.is_a?(Regexp) ? pattern.inspect[1..-2] : pattern)
-      t.regexp unless pattern.nil? # compile pattern to catch errors
-      t
-    else
-      PRegexpType::DEFAULT
-    end
+    pattern ?  PRegexpType.new(pattern) : PRegexpType::DEFAULT
```

Comment:
The string form of the `PRegexpType` is also changed in this commit to do exactly what you suggest. The above code does not "leak options into puppet land". It makes sure that such options doesn't corrupt the regexp.